codalab/lib/metadata_util.py

```diff
@@ -78,12 +77,16 @@ def request_missing_metadata(bundle_subclass, args, initial_metadata=None):
         if not new_initial_metadata[spec.key]:
             default = MetadataDefaults.get_default(spec, bundle_subclass, args)
             new_initial_metadata[spec.key] = default
-    initial_metadata = new_initial_metadata
+    return new_initial_metadata
 
-    # If args.edit exists (when doing 'cl edit'), then we want to show
-    # the editor.
+
+def request_missing_metadata(bundle_subclass, args, initial_metadata=None):
+    '''
+    For any metadata arguments that were not supplied through the command line,
+    pop up an editor and request that data from the user.
+    '''
     if not getattr(args, 'edit', True):
-        return initial_metadata
+        return fill_missing_metadata(bundle_subclass, args, initial_metadata)
```

Comment:
Should you still be calling `fill_missing_metadata` regardless of this condition?